cmake/sysbuild/partition_manager.cmake

```diff
@@ -146,8 +146,169 @@ function(partition_manager)
+    if(${part} IN_LIST IMAGES)
+      # Question, what is it we want to know ?
+      # We are now in sysbuild, meaning we know everything.
+      # So for each domain we should just include the generated hex file.
+      # Those are available thorugh sysbuild_get, but not locally as there is no parent image.
+      list(APPEND explicitly_assigned ${part})
+      sysbuild_get(${part}_PM_HEX_FILE IMAGE ${part} VAR BYPRODUCT_KERNEL_SIGNED_HEX_NAME CACHE)
+      if(NOT ${part}_PM_HEX_FILE)
+        sysbuild_get(${part}_PM_HEX_FILE IMAGE ${part} VAR BYPRODUCT_KERNEL_HEX_NAME CACHE)
+      endif()
+      set(${part}_PM_TARGET ${part})
+    else()
+      if(${part} IN_LIST containers)
+        set_ifndef(${part}_PM_HEX_FILE ${PROJECT_BINARY_DIR}/${part}.hex)
+        set_ifndef(${part}_PM_TARGET ${part}_hex)
+      endif()
+      list(APPEND implicitly_assigned ${part})
+    endif()
+  endforeach()
+
+  if (DEFINED PM_DOMAIN)
+    set(merged_suffix _${PM_DOMAIN})
+    string(TOUPPER ${merged_suffix} MERGED_SUFFIX)
+  endif()
+  set(merged merged${merged_suffix})
+  set(MERGED MERGED${MERGED_SUFFIX})
+
+  set(PM_${MERGED}_SPAN ${implicitly_assigned} ${explicitly_assigned})
+  set(${merged}_overlap TRUE) # Enable overlapping for the merged hex file.
+
+  # Iterate over all container partitions, plus the "fake" merged paritition.
+  # The loop will create a hex file for each iteration.
+  foreach(container ${containers} ${merged})
+    string(TOUPPER ${container} CONTAINER)
+
+    # Prepare the list of hex files and list of dependencies for the merge command.
+    foreach(part ${PM_${CONTAINER}_SPAN})
+      string(TOUPPER ${part} PART)
+      list(APPEND ${container}hex_files ${${part}_PM_HEX_FILE})
+      list(APPEND ${container}elf_files ${${part}_PM_ELF_FILE})
+      list(APPEND ${container}targets ${${part}_PM_TARGET})
+    endforeach()
+
+    # Do not merge hex files for empty partitions
+    if(NOT ${container}hex_files)
+      list(REMOVE_ITEM PM_${MERGED}_SPAN ${container})
+      continue()
+    endif()
+
+    # If overlapping is enabled, add the appropriate argument.
+    if(${${container}_overlap})
+      set(${container}overlap_arg --overlap=replace)
+    endif()
+
+    # Should all files still be merged ?
+    # And under which circumstances.
+    # Add command to merge files.
+    add_custom_command(
+      OUTPUT ${PROJECT_BINARY_DIR}/${container}.hex
+      COMMAND
+      ${PYTHON_EXECUTABLE}
+      ${ZEPHYR_BASE}/scripts/build/mergehex.py
+      -o ${PROJECT_BINARY_DIR}/${container}.hex
+      ${${container}overlap_arg}
+      ${${container}hex_files}
+      DEPENDS
+      ${${container}targets}
+      ${${container}hex_files}
+      # SES will load symbols from all elf files listed as dependencies to
+      # ${PROJECT_BINARY_DIR}/merged.hex. Therefore we add
+      # ${${container}elf_files} as dependency to ensure they are loaded by SES
+      # even though it is unnecessary for building the application.
```

Comment:
SES hasn't been supported in years, has it?

Reply:
let's use the opportunity to cleanup this.

SES doesn't even know what sysbuild is.